source/service.go

```diff
@@ -126,6 +127,48 @@ func NewServiceSource(ctx context.Context, kubeClient kubernetes.Interface, name
 		},
 	)
 
+	// Transformer is used to reduce the memory usage of the informer.
+	// The pod informer will otherwise store a full in-memory, go-typed copy of all pod schemas in the cluster.
+	// If watchList is not used it will not prevent memory bursts on the initial informer sync.
+	podInformer.Informer().SetTransform(func(i interface{}) (interface{}, error) {
+		pod, ok := i.(*v1.Pod)
+		if !ok {
+			return nil, fmt.Errorf("object is not a pod")
+		}
+		if pod.UID == "" {
```

Comment:
How this will behave in a scenario
- someone decided to execute `kubectl patch pod ....` and modify/change annotaions, labels or some other value of the pod?

Reply:
The informer will receive a new object that will be transformed from scratch in this case. Client-go does not alter objects after them being passed to informers

Reply:
gotcha